crates/wasi-provider/src/lib.rs

```diff
@@ -183,6 +184,12 @@ impl Provider for WasiProvider {
     fn volume_path(&self) -> Option<PathBuf> {
         Some(self.shared.volume_path())
     }
+
+    // Evict all pods upon shutdown
+    async fn shutdown(&self, node_name: &str) -> anyhow::Result<()> {
```

Comment:
@thomastaylor312 Do you think we should just never evict pods to match the official Kubelet, or are we acknowledging here that our Kubelets function differently because the tasks go away when we shut down (unlike containers which may continue running).

Reply:
The latter. Each one can do something different. I was going to change this provider later to not evict anything as we don't need to. However, your OCI provider will definitely want to do some cleanup